src/audio/drc/drc_math_generic.c

```diff
@@ -243,5 +217,39 @@ inline int32_t drc_inv_fixed(int32_t x, int32_t precision_x, int32_t precision_y
 #undef qc
 }
 
+int32_t drc_mult_lshift(int32_t a, int32_t b, int32_t lshift) { return 0; }
+int32_t drc_get_lshift(int32_t qa, int32_t qb, int32_t qy) { return 0; }
+
+#endif /* DRC_GENERIC */
+
+/*
+ * Input is Q2.30: (-2.0, 2.0)
+ * Output range: (-1.0, 1.0); regulated to Q1.31: (-1.0, 1.0)
+ */
+inline int32_t drc_sin_fixed(int32_t x)
+{
+	const int32_t PI_OVER_TWO = Q_CONVERT_FLOAT(1.57079632679489661923f, 30);
```

Comment:
"f" literal suffix means float. You get better accuracy int32_t if you omit it to treat it as double precision literal. Sriram had a similar issue.

Reply:
Thanks, I have noticed this from Sriram. I think this is worth to have another PR for the correction.

Reply:
Maybe I'm missing something but I don't see why an apparently small fix like this should wait another PR.